### PR TITLE
fix: avoid bad version of `opentelemetry-instrumentation`

### PR DESCRIPTION
**Pull request by larkee**
Version `0.23b0` of `opentelemetry-instrumentation` is causing unit tests failure relating to opentelemetry assertions. This PR updated the `google-cloud-spanner` dependency to avoid this particular version.

### CODE REVIEW

**Review by tseaver — CHANGES_REQUESTED** [review]
The main change here (in `setup.py` looks fine).

For the `testing/constraints-x.y.txt` files, the pattern in other libraries is to have the one associated with the lowest supported Python version (i.e, `testing/constraints-3.x.txt`) pin all the minimum dependencies, so that the library gets tested against them, but leave them unpinned for the higher supported versions (so that we continue to test against current versions).

**Comment by larkee** [comment]
@tseaver The test pass locally with the change but the CI is still failing :confused: Any ideas? This is blocking all PRs

**Comment by tseaver** [comment]
@larkee I'm puzzled.  Can  you tweak `noxfile` to show the installed version of `opentelemetry-instrumenation`, e.g.:

```python
    session.run("pip", "show", "opentelemetry-instrumentation")
```

I'm seeing this locally, *and* it fails:
```
Name: opentelemetry-instrumentation
Version: 0.23b2
Summary: Instrumentation Tools & Auto Instrumentation for OpenTelemetry Python
Home-page: https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/opentelemetry-instrumentation
Author: OpenTelemetry Authors
Author-email: cncf-opentelemetry-contributors@lists.cncf.io
License: Apache-2.0
Location: /home/tseaver/projects/agendaless/Google/src/python-spanner/.nox/unit-3-9/lib/python3.9/site-packages
Requires: opentelemetry-api, wrapt
Required-by: opentelemetry-sdk
```

File: setup.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "tracing": [
         "opentelemetry-api >= 1.1.0",
         "opentelemetry-sdk >= 1.1.0",
-        "opentelemetry-instrumentation >= 0.20b0",
+        "opentelemetry-instrumentation >= 0.20b0, != 0.23b0",
```
Comment (tseaver) [inline setup.py:43]:
```suggestion
        "opentelemetry-instrumentation >= 0.20b0, < 0.23dev",
```

**Review by tseaver — CHANGES_REQUESTED** [review]

**Review by tseaver — APPROVED** [review]

**Pull request merged**